### DISCUSSION TITLE
Implementation of binned observables in the Likelihood

### DISCUSSION THREAD

**Discussion by DegCo** [flav-io/flavio#225] [Q&A] [answered]
Hi all,

I have probably a simple question, but I have not found the answer in the previous issues. Hopefully it will be useful for other "rookies" like me.

When I try to implement in the likelihood binned observables just by their name, e.g. '<FL>(B0->K*mumu)', I get the error 'no measurements found'. Of course, I understand that one should also implement the binning, i.e. q2min and q2max, but I have not found a way of doing so. How do I pass in q2min and q2max information into the likelihood?

Thank you very much for your time and great work!

Example of code (that raises the error):
`FL = FastLikelihood(name='test',
                    observables= [ '<FL>(B0->K*mumu)'],
                    include_measurements=['LHCb B->K*mumu 2020 P 4-6'])
`

**Comment by peterstangl** [comment, marked as answer]
Observables that require additional arguments can be specified in three equivalent ways:
- using a tuple like e.g. `('<FL>(B0->K*mumu)', 4, 6)`
- using a list like e.g. `['<FL>(B0->K*mumu)', 4, 6]`
- using a dictionary like e.g. `{'name: '<FL>(B0->K*mumu)', 'q2min': 4, 'q2max': 6}`

**Reply by Killngai** [reply]
Dear authors,
    I also encountered the same problem as above. When I implement in the likelihood binned observables like 

`my_fastlikelihood = FastLikelihood(
name = 'my_first_test',
fit_parameters = ['C9_bsmumu'],
observables = [
('BR(Bs->mumu)'),
('BR(B0->mumu)'),
('<dBR/dq2>(B+->Kmumu)', 1.1, 6.0),
],
)`, 

it worked successfully. But When I implement another binned obs again like  
`my_fastlikelihood = FastLikelihood(
    name = 'my_first_test',
    fit_parameters = ['C9_bsmumu'],
    observables = [
                   ('BR(Bs->mumu)'),
                   ('BR(B0->mumu)'),
                   ('<dBR/dq2>(B+->Kmumu)', 1.1, 6.0),
                    ('<dBR/dq2>(B+->Kmumu)', 0.045, 6.0),
                   ],
)`
or even `('<dBR/dq2>(B+->Kee)', 1.1, 6.0),`, it throw out an error 'no measurements found'. So is there something I don't quite understand here? Thank you for telling me.

**Reply by DavidMStraub** [reply]
Well, just what is says. It doesn't find a measurement.

See here for known measurements: https://github.com/flav-io/flavio/blob/master/flavio/data/measurements.yml

**Reply by Killngai** [reply]
Thanks! It helps me a lot.  So if I want to add some new observable to `observables` argument, I must add its measurement info as well. Would you mind giving me some examples so I can learn how to add new measurements further? Even ones that didn’t exist (measurements.yml) originally.

**Reply by Killngai** [reply]
So sorry, it's me that missed this part of the information in  [add news measurements] (https://flav-io.github.io/docs/fits.html).

**Comment by DegCo** [comment]
Thank you very much! I'll make good use of it.
